ai/extract_features.py

```
@@ -114,23 +112,26 @@
 
     # Save the vectorizer
-    joblib.dump(
-        vectorizer, os.path.join(MODELS_DIR, f"{category}_tfidf_vectorizer.pkl")
-    )
+    joblib.dump(vectorizer, os.path.join(MODELS_DIR, f"{category}_tfidf_vectorizer.pkl"))
 
     # Apply SVD for dimensionality reduction
     print(f"Reducing dimensionality to {n_components} components...")
     svd = TruncatedSVD(n_components=n_components, random_state=random_state)
-    X_svd = svd.fit_transform(X_tfidf)
+    X_svd = svd.fit_transform(x_tfidf)
```

Ruff (0.11.9)

Variable `X_svd` in function should be lowercase

(N806)